pageserver/src/tenant/timeline.rs

```diff
@@ -2580,18 +2513,17 @@ impl Timeline {
                 return Err(FlushLayerError::Cancelled);
             }
 
-            guard.finish_flush_l0_layer(delta_layer_to_add.as_ref(), &frozen_layer, &self.metrics);
+            guard.finish_flush_l0_layer(&layer, &frozen_layer, &self.metrics);
 
             if disk_consistent_lsn != old_disk_consistent_lsn {
                 assert!(disk_consistent_lsn > old_disk_consistent_lsn);
                 self.disk_consistent_lsn.store(disk_consistent_lsn);
 
                 // Schedule remote uploads that will reflect our new disk_consistent_lsn
-                Some(self.schedule_uploads(disk_consistent_lsn, layers_to_upload)?)
+                Some(self.schedule_uploads(disk_consistent_lsn, [layer])?)
             } else {
                 None
             }
```

Comment:
I've raised this question earlier, is there a case where we flush an L0 layer but disk consistent lsn does not move. I think it's now clear that there is no such case.

Reply:
Perhaps we can flush empty layers after all, feels weird to add such layer to layermap but never upload it. Sounds like there should be a guard against.

Reply:
`Timeline::freeze_and_flush` is the only one which can create such empty flushes, and it is only called from tests and `import wal` functionality, which I am unsure if it is used.. `Timeline::freeze_inmem_layer` will not flush an empty layer because there should be no layer to flush in that case, should such a request arise we would hang until next flush has been completed in the `flush_frozen_layers_and_wait`, so it does not seem like this situation should ever arise.